crates/native/src/user_media.rs

```diff
@@ -1019,6 +1021,209 @@ pub struct VideoTrack {
     height: Arc<OnceLock<RwLock<i32>>>,
 }
 
+impl TrackInterface for Video {
+    /// Returns the [readyState][0] property of the underlying interface.
+    ///
+    /// [0]: https://w3.org/TR/mediacapture-streams#dfn-readystate
+    fn state(&self) -> api::TrackState {
+        self.inner.state().into()
+    }
+
+    /// Changes the [enabled][1] property of the underlying interface.
+    ///
+    /// [1]: https://w3.org/TR/mediacapture-streams#track-enabled
+    fn set_enabled(&self, enabled: bool) {
+        self.inner.set_enabled(enabled);
+    }
+}
+
+impl Video {
+    /// Adds the provided [`VideoSink`] to this [`Video`].
+    pub fn add_video_sink(&mut self, video_sink: &mut VideoSink) {
+        self.inner.add_or_update_sink(video_sink.as_mut());
+        self.sinks.push(video_sink.id());
+    }
+
+    /// Detaches the provided [`VideoSink`] from this [`Video`].
+    pub fn remove_video_sink(&mut self, mut video_sink: VideoSink) {
+        self.sinks.retain(|&sink| sink != video_sink.id());
+        self.inner.remove_sink(video_sink.as_mut());
+    }
+}
+
+impl Drop for Video {
+    fn drop(&mut self) {
+        let sink = self.sink.take().unwrap();
+        self.remove_video_sink(sink);
+    }
+}
+
+/// Representation of a [`sys::AudioTrackInterface`].
+#[derive(AsRef)]
+pub struct Audio {
```

Comment:
Let's move those structs to the `kind` module. And let's use `kind::Audio` in places where we use it. So we can clearly see that this is audio/video __kind__. Otherwise it's not clear what are those structs.